accounts/abi/bind/v2/lib_test.go

```diff
@@ -0,0 +1,393 @@
+// Copyright 2024 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package v2
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind/backends"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind/testdata/v2/events"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind/testdata/v2/nested_libraries"
```

Comment:
Whaa? This looks very strange to me.  _Actual code_ that is used in the testing itself, should IMO not reside as `testdata`.

Reply:
We decided to move this into a directory named `internal` to signal that this is not testdata that is sourced to table-driven test cases, but code that is generated and used directly from the v2 test cases.